### PR TITLE
Rust Qtum block structure fixes and test adaptation

### PR DESCRIPTION
**Pull request by lateminer**
This PR adapts block structure in `rust-qtum` crate to match Qtum Core block serialization format.

### CODE REVIEW

File: bitcoin/src/bip152.rs [hunk 1]
```diff
@@ -235,7 +235,7 @@ impl HeaderAndShortIds {
         }
 
         Ok(HeaderAndShortIds {
-            header: block.header,
+            header: block.header.clone(), // qtum: clone block header
```
Comment (alejoacosta74) [inline bitcoin/src/bip152.rs:238]: What is the point of passing a copy of the header here? This was not necessary when dealing with Bitcoin ?

File: bitcoin/src/blockdata/block.rs [hunk 2]
```diff
@@ -92,7 +92,7 @@ impl<'de> serde::Deserialize<'de> for BlockFlag {
 /// ### Bitcoin Core References
 ///
 /// * [CBlockHeader definition](https://github.com/bitcoin/bitcoin/blob/345457b542b6a980ccfbc868af0970a6f91d1b82/src/primitives/block.h#L20)
-#[derive(Copy, PartialEq, Eq, Clone, Debug, PartialOrd, Ord, Hash)]
```
Comment (alejoacosta74) [inline bitcoin/src/blockdata/block.rs:95]: Can you explain the reason for removing the `Copy` trait?

File: bitcoin/src/blockdata/block.rs [hunk 3]
```diff
@@ -109,15 +109,18 @@ pub struct Header {
...
-    pub modifier: BlockHash,
+    /// Qtum: prevoutStake
+    pub prevout_stake: OutPoint,
+    /// Qtum: vchBlockSigDlgt
+    pub signature: Vec<u8>,
     // end of qtum specific block fields
 }
 
-impl_consensus_encoding!(Header, version, prev_blockhash, merkle_root, time, bits, nonce, hash_state_root, hash_utxo_root, flags, proofhash, modifier);
+impl_consensus_encoding!(Header, version, prev_blockhash, merkle_root, time, bits, nonce, hash_state_root, hash_utxo_root, prevout_stake, signature);
```
Comment (alejoacosta74) [inline bitcoin/src/blockdata/block.rs:123]: Have you tested the macro can succesfully deserialize the fields `prevout_stake` and `signature`?

**Review by alejoacosta74 — COMMENTED** [review]
This looks like a good step forward. Can you answer the open questions?

File: bitcoin/src/bip152.rs (hunk 1 above)
Comment (lateminer) [reply bitcoin/src/bip152.rs:238]: The reason behind that is that Qtum block header contains block signature, and its `Vec<u8>` type doesn't implement `Copy` trait.

File: bitcoin/src/blockdata/block.rs (hunk 2 above)
Comment (lateminer) [reply bitcoin/src/blockdata/block.rs:95]:
```
pub signature: Vec<u8>,
    |     ---------------------- this field does not implement `std::marker::Copy`
```
Probably there is a better way to solve this.

File: bitcoin/src/blockdata/block.rs (hunk 3 above)
Comment (lateminer) [reply bitcoin/src/blockdata/block.rs:123]: Not yet. However, it might be a good idea to write a separate test for that. I will have a look.

Comment (lateminer) [reply bitcoin/src/blockdata/block.rs:123]: Done: https://github.com/qtumproject/rust-qtum/pull/1/commits/22f344ab7069cc061fe8f5eeba71758929cf5e72
It works.

**Pull request merged**